serveradmin/dataset/commit.py

```diff
@@ -24,45 +24,60 @@ def __init__(self, commit, user):
         self.user = user
 
         if 'deleted' in commit:
-            self.deleted_servers = commit['deleted']
+            self.deleted = commit['deleted']
             self._validate_deletes()
         else:
-            self.deleted_servers = []
+            self.deleted = []
 
         if 'changes' in commit:
-            self.changed_servers = commit['changes']
+            self.changes = commit['changes']
             self._decorate_changes()
             self._validate_changes()
             self._clean_changes()
         else:
-            self.changed_servers = {}
+            self.changes = {}
 
-        # TODO Only use Django objects
-        self.servers = _fetch_servers(self.changed_servers)
+        self.changed_servers = _fetch_servers(self.changes.keys())
+        self.deleted_servers = _fetch_servers(self.deleted)
 
         # If non-empty, the commit will go through the backend, but an error
         # will be shown on the client.
         self.warnings = []
 
+    def access_control(self, action, servers):
+        if not self.user or self.user.is_superuser:
+            return
+
+        access_control_groups = self.user.access_control_groups.all()
+        for server in servers.values():
```

Comment:
I never learned docstring syntax.  I don't think it is a good investment to learn it now.  Python 3.5 has optional datatype safety [1].  We still have 3.4 on production, though.  We can wait a little longer and start using it.

[1] https://docs.python.org/3/library/typing.html